Firebase/Auth/Source/FIRAuth.m

```diff
@@ -1092,6 +1128,23 @@ - (void)phoneNumberSignInWithRequest:(FIRVerifyPhoneNumberRequest *)request
 }
 #endif
 
+/** @fn internalSignInAnonymouslyWithCompletion:
+    @param completion A block which is invoked when the anonymous sign in request completes.
+ */
+- (void)internalSignInAnonymouslyWithCompletion:(FIRSignupNewUserCallback)completion {
+  FIRSignUpNewUserRequest *request =
+      [[FIRSignUpNewUserRequest alloc]initWithRequestConfiguration:_requestConfiguration];
+  [FIRAuthBackend signUpNewUser:request
+                       callback:^(FIRSignUpNewUserResponse *_Nullable response,
+                                  NSError *_Nullable error) {
+    if (error) {
+      completion(nil, error);
+      return;
+    }
+    completion(response, nil);
+  }];
```

Comment:
Looks like the callback block is equivalent of just calling `completion(response, error)`. If so, you can just pass `completion` to `FIRAuthBackend signUpNewUser:callback:` directly.

Reply:
Great catch! done.